### PR TITLE
Allow the user to specify a NGINX port

### LINKED ISSUE #253: Allow NGINX to run on a custom port
I have my own RP running on port 80/443 already that handles all SSL termination.

When umbrel starts up it tries to bind to port 80 which is already taken.

Maybe some option to stop Nginx binding to the host's port 80 would be helpful. 

For now, I could work around it by commenting out these lines in docker-compose:
```
                # ports:
                #     - "80:80"
```

### PR DESCRIPTION
**Pull request by nolim1t**
This issue addresses and fixes #253

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ services:
                 restart: unless-stopped
                 stop_grace_period: 30s
                 ports:
-                    - "80:80"
+                    - "${WEBPORT:-80}:80"
```
Comment (lukechilds) [inline docker-compose.yml:33]:
```suggestion
                    - "${UMBREL_HTTP_PORT:-80}:80"
```

**Review by lukechilds — CHANGES_REQUESTED** [review]
Thanks @nolim1t this looks great!

Just added a suggested change for the env var to keep it inline with the style of the existing ones.

File: docker-compose.yml (hunk 1 above)
Comment (nolim1t) [reply docker-compose.yml:33]: Ok

**Comment by mayankchhabra** [comment]
Maybe we should add the new variable to ".env" for it to persist between OTA updates.

**Comment by lukechilds** [comment]
Yeah good point.

@nolim1t You can see examples in the configure script of how to persist settings in `.env`.

**Comment by nolim1t** [comment]
> Maybe we should add the new variable to ".env" for it to persist between OTA updates.

I allowed for both. Default to 80 if not set.

**Comment by lukechilds** [comment]
If users modify the `.env` file directly, it'll be wiped after an OTA update.

See an example of how this needs to be done in the configure script.

Or see this PR for an example of how we added `BITCOIN_RPC_AUTH `: https://github.com/getumbrel/umbrel/pull/255/files

**Comment by nolim1t** [comment]
Hmm what's the point of the .env file if you are wiping it each time.

May as well just keep it in the compose file 😛 

Better to just add what's needed and use default values if nothing is set.

**Comment by nolim1t** [comment]
I've added `UMBREL_HTTP_PORT` to the configure script too.. it will [persist to `.env' if this set](https://github.com/getumbrel/umbrel/pull/256/commits/df7dbcea55de191add284504b47f1aa66ceb3484). Otherwise it will just stick with the defaults as per docker-compose.yml config.

**Comment by nolim1t** [comment]
Also added how to use this setting to the README file

File: scripts/configure [hunk 2]
```diff
@@ -198,6 +198,11 @@ if [[ -f "${STATUS_DIR}/node-status-bitcoind-ready" ]]; then
   sed -i "s/bitcoin.node=.*/bitcoin.node=bitcoind/g;" "$LND_CONF_FILE"
 fi
 
+# If there was an UMBREL_HTTP_PORT set before then add it to the $ENV_FILE
+if [ ! -z $UMBREL_HTTP_PORT ]; then
```
Comment (lukechilds) [inline scripts/configure:202]: This will throw an unbound variable error if `UMBREL_HTTP_PORT` is not set.

**Review by lukechilds — CHANGES_REQUESTED** [review]

File: scripts/configure (hunk 2 above)
Comment (nolim1t) [reply scripts/configure:202]: ok ill remove it

**Review by mayankchhabra — CHANGES_REQUESTED** [review]
Thanks, @nolim1t. 

It'd be great to stick to using the same pattern that we use to set environment variables in `.env` instead of echoing it directly, i.e. by having `UMBREL_HTTP_PORT=<port>` in [`templates/.env-sample`](https://github.com/getumbrel/umbrel/blob/master/templates/.env-sample) and setting it via a `sed` in the config script.

**Comment by nolim1t** [comment]
> Thanks, @nolim1t.
> 
> It'd be better if we can stick to the same pattern that we use to set environment variables in `.env` instead of echoing it directly, i.e. by having `UMBREL_HTTP_PORT=<port>` in [`templates/.env-sample`](https://github.com/getumbrel/umbrel/blob/master/templates/.env-sample) and setting it via a `sed` in the config script.

does the following changes work?

**Review by louneskmt — COMMENTED** [review]
You've set NGINX_PORT in the `.env` file but UMBREL_HTTP_PORT in all the other files.

**Comment by nolim1t** [comment]
Thanks well spotted!

File: README.md [hunk 3]
```diff
@@ -57,6 +57,9 @@ sudo NETWORK=testnet ./scripts/start
 
 # For regtest, run:
 sudo NETWORK=regtest ./scripts/start
+
+# For a custom HTTP_PORT (example 8080)
+sudo UMBREL_HTTP_PORT=8080 ./scripts/start
```
Comment (louneskmt) [inline README.md:62]:
```suggestion
sudo NGINX_PORT=8080 ./scripts/start
```

File: docker-compose.yml [hunk 4]
```diff
@@ -30,7 +30,7 @@ services:
                 restart: on-failure
                 stop_grace_period: 30s
                 ports:
-                    - "80:80"
+                    - "${UMBREL_HTTP_PORT:-80}:80"
```
Comment (louneskmt) [inline docker-compose.yml:33]:
```suggestion
                    - "${NGINX_PORT:-80}:80"
```

**Review by louneskmt — CHANGES_REQUESTED** [review]
There were two left 😁

**Comment by nolim1t** [comment]
Thanks again lol

**Comment by maltokyo** [comment]
Awesome to see this coming! Finally, support for being behind a reverse proxy.

**Comment by nolim1t** [comment]
> Awesome to see this coming! Finally, support for being behind a reverse proxy.

yep just need someone to review it and we are good

**Comment by junderw** [comment]
I've added NGINX_PORT to my .env file in anticipation for this update :-D

**Comment by lukechilds** [comment]
Closed by https://github.com/getumbrel/umbrel/pull/562

**Pull request closed without merging**